test/src/test/java/org/apache/accumulo/test/InMemoryMapIT.java

```diff
@@ -0,0 +1,320 @@
+    assertMutatesEquivalent(mutations, defaultMap, localityGroupMapWithNative);
+  }
+
+  /**
+   * Assert that a set of mutations mutate to equivalent map in both of the InMemoryMaps.
+   * <p>
+   * In this case, equivalent means 2 things.
+   * <ul>
+   * <li>The size of both maps generated is equal to the number of key value pairs in all mutations passed</li>
+   * <li>The size of the map generated from the first InMemoryMap equals the size of the map generated from the second</li>
+   * <li>Each key value pair in each mutated map has a unique id (kvCount)</li>
+   * </ul>
+   *
+   * @param mutations
+   *          List of mutations
+   * @param imm1
+   *          InMemoryMap to compare
+   * @param imm2
+   *          InMemoryMap to compare
+   */
+  private void assertMutatesEquivalent(List<Mutation> mutations, InMemoryMap imm1, InMemoryMap imm2) {
+    int mutationKVPairs = countKVPairs(mutations);
+
+    List<MemKey> memKeys1 = getArrayOfMemKeys(imm1);
+    List<MemKey> memKeys2 = getArrayOfMemKeys(imm2);
+
+    assertEquals("Not all key value pairs included: " + dumpInMemoryMap(imm1, memKeys1), mutationKVPairs, memKeys1.size());
+    assertEquals("Not all key value pairs included: " + dumpInMemoryMap(imm2, memKeys2), mutationKVPairs, memKeys2.size());
+    assertEquals("InMemoryMaps differ in size: " + dumpInMemoryMap(imm1, memKeys1) + "\n" + dumpInMemoryMap(imm2, memKeys2), memKeys1.size(), memKeys2.size());
+    assertEquals("InMemoryMap did not have distinct kvCounts " + dumpInMemoryMap(imm1, memKeys1), mutationKVPairs, getUniqKVCount(memKeys1).size());
+    assertEquals("InMemoryMap did not have distinct kvCounts " + dumpInMemoryMap(imm2, memKeys2), mutationKVPairs, getUniqKVCount(memKeys2).size());
+
+  }
+
+  private int countKVPairs(List<Mutation> mutations) {
+    int count = 0;
+    for (Mutation m : mutations) {
+      count += m.size();
+    }
+    return count;
+  }
+
+  private List<MemKey> getArrayOfMemKeys(InMemoryMap imm) {
+    SortedKeyValueIterator<Key,Value> skvi = imm.compactionIterator();
+
+    List<MemKey> memKeys = new ArrayList<MemKey>();
+    try {
+      skvi.seek(new Range(), new ArrayList<ByteSequence>(), false); // everything
+      while (skvi.hasTop()) {
+        memKeys.add((MemKey) skvi.getTopKey());
+        skvi.next();
+      }
+    } catch (IOException ex) {
+      java.util.logging.Logger.getLogger(InMemoryMapIT.class.getName()).log(Level.SEVERE, null, ex);
+    }
+
+    return memKeys;
+  }
+
+  private String dumpInMemoryMap(InMemoryMap map, List<MemKey> memkeys) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("InMemoryMap type ");
+    sb.append(map.getMapType());
+    sb.append("\n");
+
+    for (MemKey mk : memkeys) {
+      sb.append("  ");
+      sb.append(mk.toString());
+      sb.append("\n");
+    }
+
+    return sb.toString();
+  }
+
+  private List getUniqKVCount(List<MemKey> memKeys) {
+    List<Integer> kvCounts = new ArrayList<Integer>();
```

Comment:
Nevermind, i see the set at the end.  Why not just return a count instead of a list, thats what the name implies?  Returning the list was misleading for me, because the type does not feel unique.